src/plugins/intel_gna/tests/unit/gna_export_scores_test.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <map>
+#include <vector>
+
+#include "any_copy.hpp"
+#include "common_test_utils/data_utils.hpp"
+#include "common_test_utils/ngraph_test_utils.hpp"
+#include "gna_data_types.hpp"
+#include "gna_plugin.hpp"
+#include "pre_post_process/preprocessing.hpp"
+
+using namespace InferenceEngine;
+
+namespace testing {
+
+class GNAPluginForOutPrecisionTest : public ov::intel_gna::GNAPlugin {
+public:
+    using GNAPlugin::GNAPlugin;
+    void setGNADeviceHelper() {
+        gnadevice = std::make_shared<GNADeviceHelper>();
+    }
+
+    bool isGnaDevicePresent() const {
+        return gnadevice.get() != nullptr;
+    }
+
+    void setAvx2Support(bool testAvx2) {
+        isAvx2Supported = testAvx2;
+    }
+
+    bool isAvx2Support() const {
+        return isAvx2Supported;
+    }
+};
+
+typedef std::tuple<InferenceEngine::Precision,   // input precision
+                   InferenceEngine::Precision,   // output precision
+                   InferenceEngine::SizeVector,  // input shape
+                   intel_dnn_orientation_t,      // orientation
+                   ov::AnyMap,                   // gna config
+                   bool,                         // test avx2
+                   uint32_t                      // input range
+                   >
+    GNAOutputPrecisionParams;
+
+template <typename U, typename T>
+class GNAOutputPrecisionTest : public ::testing::TestWithParam<GNAOutputPrecisionParams> {
+public:
+    void SetUp() override {
+        ov::AnyMap gna_config;
+        uint32_t input_range;
+        std::tie(precisionIn, precisionOut, shape, orientation, gna_config, testAvx2, input_range) = GetParam();
+        if (!std::is_same<T, U>::value) {
+            sf = gna_config[ov::intel_gna::scale_factors_per_input.name()].as<std::map<std::string, float>>()["0"];
+        }
+        inputValues.resize(ov::shape_size(shape));
+        CommonTestUtils::fill_data_random(&inputValues[0], ov::shape_size(shape), input_range);
+
+        std::transform(begin(inputValues), end(inputValues), std::back_inserter(referenceValues), [this](U i) {
+            return static_cast<T>(i / sf);
+        });
+
+        plugin.reset(new GNAPluginForOutPrecisionTest(any_copy(gna_config)));
+#ifdef HAVE_AVX2
```

Comment:
in propsed changes above whole control would be done from factory level, you would not need to change it here.